src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java

```diff
@@ -316,7 +316,8 @@ public void saveAs() throws Exception {
      */
     public void saveAs(File file) throws Exception {
         BibDatabaseContext context = panel.getBibDatabaseContext();
-
+        Path oldFile = context.getDatabasePath().get();
```

Comment:
Not your fault: I always wondered why this is an Optional, I think in this case I see no reason why it could be empty, so you are probably on the save side calling directly get.